src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -58,7 +58,7 @@ public class OutputConsoleLogger : INuGetUILogger, IDisposable
 
             ErrorListTableDataSource = errorListDataSource;
 
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            _outputConsole = AsyncLazy.New(async () =>
```

Comment:
We're comfortable delaying this because there's no need to subscribe to build events that would affect our logging, if we haven't logged anything in the first place. 

Is that a good summary?

Reply:
OK, got me thinking...

Logger is created at possibly two places:
- `NuGetUIFactory` has `[Import]` property and `[ImportingConstructor]`.
- `GeneralOptionControl` gets it through `ServiceLocator`.

When Visual Studio is started, and solution is opened, NuGetPackage gets instantiated, which will create NuGetUIFactory, which will, in turn, create OutputConsoleLogger, so that first _ever_ (in lifetime of devenv.exe) logging operation will create OutputConsole and hook up operations. We do not need to listen to build start before that, because there will be no NuGet errors in error list, and we will get all future build starts so we will have chance to clear up error list.

So, it seems we are good. 🙂 I will, off course, do some manual testing.